client/src/components/ColorList.js

```diff
@@ -18,15 +23,43 @@ const ColorList = ({ colors, updateColors }) => {
 
   const saveEdit = e => {
     e.preventDefault();
-    // Make a put request to save your updated color
-    // think about where will you get the id from...
-    // where is is saved right now?
+    updateColor(colorToEdit)
   };
 
+  const handleNewColor = e => {
+    e.preventDefault();
+    addColor();
+  }
+
+  const updateColor = () => {
+    axiosWithAuth()
+      .put(`/api/colors/${colorToEdit.id}`, colorToEdit)
+      .then(() => {
+        fetchColor()
+      })
+      .catch(error => console.log(error))
+      setColorToEdit({color: "", code: { hex: "" }})
+  }
+
```

Comment:
I'd move the form reset inside the `.then()` statement. That way the form will only reset if the update is successful.